src/book.py

```diff
@@ -0,0 +1,224 @@
+
+        if lemmatize:
+            WNLemma = WordNetLemmatizer()
+            # The Lemmatizer only takes single words as inputs, so we need to
+            # break our text into individual words
+            tokens = word_tokenize(cleaned)
+            if lemma_pos != 'all':
+                lemmatized = [WNLemma.lemmatize(
+                    token, pos=lemma_pos) for token in tokens]
+            else:
+                lemmatized = [WNLemma.lemmatize(WNLemma.lemmatize(
+                    WNLemma.lemmatize(token, pos='n'), pos='v'), pos='a')
+                    for token in tokens]
+            cleaned = (" ").join(lemmatized)
+
+        if deromanize:
+            tokens = word_tokenize(cleaned)
+            regex_roman = '^(?=[MDCLXVI])M*(C[MD]|D?C{0,3})(X[CL]|L?X{0,3})(I[XV]|V?I{0,3})$'
+            cleaned = [word for word in tokens if re.search(
+                rex_roman, word, flags=re.IGNORECASE) is None]
+
+        if inplace:
+            self._text = cleaned
+        else:
+            return BookText(filepath=None, rawtext=cleaned, author=self._author)
+
+    def tokenize(self, on, rem_stopwords=True, stopword_lang='english',
+                 add_stopwords=[]):
+        """Tokenize words or sentences in the text
+
+        Produces lists of either words or sentences contained in the text
+
+        **NB** converts words to lower case to facilitate comparisons
+
+        on ('word' or 'sentence'):
+            whether the lists will be tokenized according to words or sentences
+        rem_stopwords (default True): if stopwords should be removed from tokens
+        stopword_lang (default 'english'): language of stopword corpus to use
+        add_stopwords CURRENTLY UNWORKING (default []): list of words to be added to stopword list
+        """
+        if 'word' in on.lower():
+            token = word_tokenize(self._text.lower())
+        elif 'sent' in on.lower():
+            token = sent_tokenize(self._text.lower())
+        else:
+            raise KeyError(
+                "Arugument 'on' must refer to either word or sentence")
+        if rem_stopwords:
+            stop_words = set(stopwords.words(stopword_lang))
+            # Additional words can be added to the stop word list
+            # stop_words.extend(add_stopwords)
+            if 'word' in on.lower():
+                token = [word for word in token if not word in stop_words]
+            elif 'sent' in on.lower():
+                # TODO: Is there a better way to do this?
+                for i in range(len(token)):
+                    sent = token[i]
+                    words = sent.split()
+                    words_nostop = [
+                        word for word in words if not word in stop_words]
+                    token[i] = (" ").join(words_nostop)
+        return token
+
+    def snippet(self, length, on, non_cont=False, inplace=False):
+        """
+        """
+        # TODO add ability for random seed
```

Comment:
We must also be able to generate the same snippet repeatedly if we want (for testing). Add the functionality for a random_state alongside.

Reply:
This is already mentioned in the `# TODO` comment, and I think can be saved for a future commit. The goal here is to get the ball rolling so everyone has the same set of tools as quickly as possible.

Reply:
Right, I missed the `# TODO` comment. I just wanted to put this as a reminder.